tests/integration/cmd_devfile_deploy_test.go

```diff
@@ -108,34 +103,6 @@ var _ = Describe("odo devfile deploy command tests", func() {
 					session.WaitEnd()
 				})
 
-				When("deleting previous deployment and switching kubeconfig to another namespace", func() {
```

Comment:
This might be an overkill, but I think you can add a test case where:
```go
When "env.yaml containing project is left behind":
    When "current namespace is different from env.yaml project":
        When "odo deploy is run":
            It "should take the current namespace into account"
```

A similar test for `odo delete component`, perhaps.